django_bmemcached/memcached.py

```diff
@@ -38,10 +39,7 @@ def _cache(self):
             return client
 
         if self._options:
-            client = self._lib.Client(
-                self._servers, self._options.get('username', None),
-                self._options.get('password', None)
-            )
+            client = self._lib.Client(self._servers, **self._options)
```

Comment:
Post Django 1.11, the `CACHES` configuration may contain an `OPTIONS` dict of additional memcached settings that can be passed through to the client. https://docs.djangoproject.com/en/dev/releases/1.11/#cache

> Memcached backends now pass the contents of OPTIONS as keyword arguments to the client constructors, allowing for more advanced control of client behavior. See the cache arguments documentation for examples.

If options are passed in, we want to pass them all through - not extract username / password and reject the rest (which is what this currently does).

Reply:
Hey there, I think this could be a problem because it could send unknown keyword args to `Client` for example, in your previous configuration it would break.
What do you think of creating a set with the allowed configurations and make a warning for non-existent ones so users' codebase won't break on the update?
I thought of something like this:
```python
import pickle
import warnings

ALLOWED_OPTIONS = {
    'username',
    'password',
    'compression',
    'socket_timeout',
    'pickler',
    'unpickler',
    'pickle_protocol',
}

self_options = {
    'username': 'user',
    'password': 'password',
    'compression': None,
    'socket_timeout': 3,
    'pickler': pickle.Pickler,
    'unpickler': pickle.Unpickler,
    'pickle_protocol': 0,
    'unknown': 'key',
    'wrong': 'config'
}

new_options = {valid_option: self_options[valid_option] for valid_option in ALLOWED_OPTIONS & self_options.keys()}
wrong_options = self_options.keys() - ALLOWED_OPTIONS
if wrong_options:
    warnings.warn(
        'BMemached was initialized with these invalid options, they will be ignored {!r}'.format(wrong_options))

# Construct Client with **new_options
```
the result:
```
/Users/jayson/p/personal/django-bmemcached/venv/bin/python /Users/jayson/Library/Preferences/PyCharm2019.2/scratches/scratch.py
/Users/jayson/Library/Preferences/PyCharm2019.2/scratches/scratch.py:29: UserWarning: BMemached was initialized with these invalid options, they will be ignored {'unknown', 'wrong'}
  warnings.warn('BMemached was initialized with these invalid options, they will be ignored {!r}'.format(wrong_options))
```

Reply:
Sure - I actually ran into this myself. My initial thought here was that although it blows up, it's up to the user to ensure that they are passing valid options - however, it's unreasonable to expect users to look at the source code to work out what they are - this would provide a better experience. Will update.

Reply:
One question about this - given that there is a 1:1 correlation between the memcache client and options in the CACHES config, is there any valid reason for accepting invalid options? Should this raise an error (i.e. fail hard and fast in the event of invalid options).